ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -125,4 +216,191 @@ mod tests {
         };
         check(&nd, &mut rng, 1_u8);
     }
+
+    /// Tests for Geometric
+    #[test]
+    fn test_geometric_constructor() {
+        // should fail for negative probability
+        let p = -1.0;
+        let expected = Err(Error::BadGeometricProb(p));
+        let actual = Geometric::new(p);
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_geometric_sample_dist() {
+        let mut rng = rand::thread_rng();
+        let p = 0.5; // success probability
+        let geometric = Geometric::new(p).expect("Geometric not constructed");
+        let mut histogram = HashMap::new();
+        let num_samples = 100_000;
+        for _ in 0..num_samples {
+            let sample = geometric.sample(&mut rng);
+            *histogram.entry(sample).or_insert(0) += 1;
+        }
+        #[allow(clippy::cast_precision_loss)]
+        for x in 0..100 {
+            let observed_probability = histogram
+                .get(&x)
+                .map_or(0.0, |count| f64::from(*count) / f64::from(num_samples));
+            let expected_probability = (1.0 - p).powf(f64::from(x)) * p;
+            println!("x = {x}, Observed Probability = {observed_probability}, Expected Probability = {expected_probability}");
+            assert!((observed_probability - expected_probability) <= 0.01);
+        }
+    }
+    /// Tests for Double Geometric
+    #[test]
+    fn test_double_geometric_constructor() {
+        // should fail for negative s parameter
+        let s = -1.0;
+        let n = 25;
+        let expected = Err(Error::BadS(s));
+        let actual = DoubleGeometric::new(s, n);
+        assert_eq!(expected, actual);
+    }
+    /// Tests for Truncated Double Geometric
+    #[test]
+    fn test_truncated_double_geometric_constructor() {
+        // should fail for negative s parameter
+        let mut s = -1.0;
+        let mut n = 25;
+        let mut expected = Err(Error::BadS(s));
+        let mut actual = TruncatedDoubleGeometric::new(s, n);
+        assert_eq!(expected, actual);
+        s = 2.0;
+        n = 3_000_000;
+        expected = Err(Error::BadShiftValue(n));
```

Comment:
Use literals for expectations.  (Generally, you should avoid mut variables, get too accustomed to mutability and you will make mistakes.)

Reply:
not quite sure what you're looking for by literals here. Something like `assert_eq!(Err(Error::BadS(s)), actual);`

Reply:
I'm suggesting that instead of

```rust
let expected = foo;
assert_eq!(expected, bar);
```

You just do this:

```rust
assert_eq!(foo, bar);
```

Reply:
okay, I'll update to that here and these related tests